core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -107,6 +107,14 @@ class SparkHadoopUtil extends Logging {
         if (key.startsWith("spark.hadoop.")) {
           hadoopConf.set(key.substring("spark.hadoop.".length), value)
         }
+	// Copy any "fs.swift2d.foo=bar" properties into conf as "fs.swift2d.foo=bar"
```

Comment:
My intention to filter out Swift and S3 specific properties and add them to new Hadoop conf is just to make sure that my changes are not impacting existing spark behavior for Hadoop interaction and add more functionality for S3 and swift. SparkHadoopUtil.get.conf is used many other places for reading data specifically from Hadoop, if user updates some config for Hadoop and unintentionally he may cause some disruptive behavior. I wanted to keep the impact of changes as minimum as possible.